### ISSUE TITLE
Incorrect parsing of Q_PROPERTY

### ISSUE THREAD

**Issue by kossebau** [doxygen/doxygen#7734]
Q_PROPERTY(unsigned long propertyName READ [...]) -> parsed as type "unsigned long propertyName", not "unsigned long"

Doxygen 1.8.16 (but also versions before) has issues parsing Qt's declarations of class properties which use the type name `unsigned long`.
Example (taken from KCoreAddons' KJob class from the KDE Frameworks):
```
    Q_PROPERTY(unsigned long percent READ percent NOTIFY percent)
```
results in the entry in the class "Properties" section:
```
unsigned long percent  READ
```
where it should have been
```
unsigned long  percent
```
Workaround seems to use the typedef provided by Qt's qglobal.h header, `ulong` or using the C++ long name variant `unsigned long int`.

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

The current version of doxygen is 1.8.18, did / can you try to run this version as well?

**Comment by kossebau** [comment]
Build doxygen from current master now, and bug is still present.

To reproduce:

1. Place file named "object.h" in an empty directory (see content below)
2. There run "doxygen -g" & "doxygen Doxyfile"
3. Browse html/classObject.html

Header file content (embedded, because github seems to not support attaching .h files?):
```cpp
#include <QObject>

/// A class with a property of type "unsigned long", noted in that type name variant
class Object : public QObject
{
    Q_OBJECT
    /// Doxygen reads here a type "unsigned long foo" and name "READ", not "unsigned long" & "foo"
    Q_PROPERTY(unsigned long foo READ foo NOTIFY fooChanged)
public:
    /// Constructor
    Object();
    /// @returns foo
    unsigned long foo() const;
Q_SIGNALS:
    /// Emitted when foo has changed
    void fooChanged(unsigned long foo);
};
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7742

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by kossebau** [comment]
Thanks for having worked on this :) (sorry, not yet gotten around to test the patch, defaulting to assume your fix is perfect :) if I get to it any next day, will still try to report back here just to assure))

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
